src/tests/test_base_tensorflow_model.py

```diff
@@ -36,59 +36,65 @@ def test_get_model_blocks(self):
         self.assertTrue((weights1 == weights2).all())
 
     def test_model_get_loss(self):
-        modelLoss = self.nonTrainableModel.get_loss(outputs=self.nonTrainableExpectedOutput, targets=self.exampleTargets)
+        modelLoss = self.nonTrainableModel.get_loss(
+            targets=self.exampleTargets,
+            training=False,
+            **self.exampleInputs
+        )
         self.assertEqual(modelLoss.numpy(), self.expectedLoss)
 
-    def test_model_train(self):
-        self.trainableModel(inputs=self.exampleInputs)  # first call initializes layers and weights
-        weights1 = self.trainableModel.exampleBlock.exampleDenseLayer.weights[0].numpy()
-        self.trainableModel.train(inputs=self.exampleInputs, targets=self.exampleTargets, optimizer=tf.keras.optimizers.Adam(lr=1))
-        weights2 = self.trainableModel.exampleBlock.exampleDenseLayer.weights[0].numpy()
-        self.assertNotEqual(weights1, np.array([]))
-        self.assertNotEqual(weights2, np.array([]))
-        self.assertFalse((weights1 == weights2).all())
-
     def test_model_produce_metrics(self):
-        metrics = self.nonTrainableModel.produce_metrics(inputs=self.exampleInputs, outputs=self.nonTrainableExpectedOutput, targets=self.exampleTargets)
+        metrics = self.nonTrainableModel.produce_metrics(
+            targets=self.exampleTargets,
+            training=False,
+            globalStep=100,
+            **self.exampleInputs
+        )
         self.assertTrue((metrics.numpy() == self.exampleInputs['input1'].numpy()).all())
 
-    def test_non_trainable_model_evaluate(self):
-        outputs = self.nonTrainableModel(inputs=self.exampleInputs, training=False)
-        modelLoss = self.nonTrainableModel.get_loss(outputs=self.nonTrainableExpectedOutput, targets=self.exampleTargets)
-        metrics = self.nonTrainableModel.produce_metrics(inputs=self.exampleInputs, outputs=self.nonTrainableExpectedOutput, targets=self.exampleTargets)
-        evaluationOutput = self.nonTrainableModel.evaluate(inputs=self.exampleInputs, targets=self.exampleTargets)
-        self.assertTrue((evaluationOutput[0].numpy() == outputs.numpy()).all())
-        self.assertTrue(evaluationOutput[1] == modelLoss.numpy())
-        self.assertTrue((evaluationOutput[2].numpy() == metrics.numpy()).all())
-
-    def test_trainable_model_evaluate(self):
-        self.trainableModel(inputs=self.exampleInputs)
-        weights1 = self.trainableModel.exampleBlock.exampleDenseLayer.weights[0].numpy()
-        self.nonTrainableModel.evaluate(inputs=self.exampleInputs, targets=self.exampleTargets)
-        weights2 = self.trainableModel.exampleBlock.exampleDenseLayer.weights[0].numpy()
-        self.assertNotEqual(weights1, np.array([]))
-        self.assertNotEqual(weights2, np.array([]))
-        self.assertTrue((weights1 == weights2).all())
+    # TODO(Laszlo) no model.evaluate()
+    # def test_non_trainable_model_evaluate(self):
```

Comment:
these need to be rewritten because the API is 100% changed, didn't took it out because easier to rewrite